### PR TITLE
AB#8516,8515 - Fix add events erasing previous state. Connect drag-drop to state.

### PR DESCRIPTION
**Pull request by sc-antonisnikolaidis**

### CODE REVIEW

File: play-media-native/features/SportPicker/SportPicker.tsx [hunk 1]
```diff
@@ -43,6 +43,16 @@ export const SportPicker = ({ onChange, sports, initialValue = '' }: SportPicker
     [onChange]
   );
 
+  const isSelected = useCallback(
```
Comment (chpa7) [inline play-media-native/features/SportPicker/SportPicker.tsx:46]: SportPicker looks broken on Create Athlete screen 
image

**Review by chpa7 — CHANGES_REQUESTED** [review]

**Comment by chpa7** [comment]
@sc-nelikostadinova Feel free to review as well

**Comment by sc-nelikostadinova** [comment]
There is still the issue of the double highlighting of sports when editing (both in athletes and events) but I don't know if it's supposed to be fixed in this PR or not.

**Review by sc-nelikostadinova — APPROVED** [review]

**Comment by sc-antonisnikolaidis** [comment]
@sc-nelikostadinova @chpa7 Sport Picker looks fine on my Android. Unfortunately another iOS thing.

File: play-media-native/screens/CreateEvent/FieldsView.tsx [hunk 2]
```diff
@@ -1,12 +1,67 @@
...
+  setDate,
+  setTitle,
+  setLocation,
+  setShowDatePicker,
+  showDatePicker,
+  sport,
+  sports,
+  title,
+}) => {
+  const openDatePicker = useCallback(() => {
```
Comment (chpa7) [inline play-media-native/screens/CreateEvent/FieldsView.tsx:23]: The DatePicker is always open now, and can't be dismissed by clicking on cancel or OK

image

**Review by chpa7 — CHANGES_REQUESTED** [review]

File: play-media-native/screens/CreateEvent/FieldsView.tsx (hunk 2 above)
Comment (sc-antonisnikolaidis) [reply play-media-native/screens/CreateEvent/FieldsView.tsx:23]: fixed

File: play-media-native/features/SportPicker/SportPicker.tsx (hunk 1 above)
Comment (chpa7) [inline play-media-native/features/SportPicker/SportPicker.tsx:46]: Is this needed? 

Why not just 

`
const [value, setValue] = useState(initialValue);
`
.....
`<RadioButton.Group onValueChange={handleChange} value={value}>
`
.....
`
backgroundColor:
   value === sport.title ? theme.colors.yellow.DEFAULT : theme.colors.white.DEFAULT
`

that is easier to read?

**Review by chpa7 — APPROVED** [review]

**Pull request merged**